src/databricks/labs/ucx/source_code/linters/spark_connect.py

```diff
@@ -22,6 +22,15 @@ def _cluster_type_str(self) -> str:
     def lint(self, node: ast.AST) -> Iterator[Advice]:
         pass
 
+    def lint_tree(self, tree: ast.AST) -> Iterator[Advice]:
+        reported_locations = set()
+        for node in ast.walk(tree):
+            for advice in self.lint(node):
+                loc = (advice.start_line, advice.start_col)
+                if loc not in reported_locations:
+                    reported_locations.add(loc)
+                    yield advice
```

Comment:
```suggestion
                if loc in reported_locations:
                    continue
                reported_locations.add(loc)
                yield advice
```
break the loops early for a more readable flow.